compiler/optimizer/VPHandlers.cpp

```diff
@@ -12272,7 +12340,21 @@ TR::Node *constrainArrayStoreChk(OMR::ValuePropagation *vp, TR::Node *node)
                if (isInstance == TR_yes)
                   {
                   vp->registerPreXClass(object);
-                  canBeRemoved = true;
+
+                  // Arrays with value type component types must throw NPE if
+                  // the value assigned is null.  Can only remove the check if
+                  // the array cannot have a value type as its component type
+                  // or the value cannot be a null reference
+                  //
+                  // VALHALLA_TODO:  For case where array component type is
```

Comment:
This should be VALUE_TYPE_TODO not VALHALLA since we are talking about the language agnostic implementation here which OpenJ9 may want/need to specialize.

Reply:
Fixed in https://github.com/eclipse/omr/commit/71c19560f51e0846bbeeed2703fd861800a38c7d/